### PR TITLE
fix: fork app button not opening the fork modal for non-signed in users

### LINKED ISSUE #11745: [Bug]: Forking public app via new user sign-up results in a 400 response on prod
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description
Moving original description below since the behavior on release and prod is a little different when trying to fork public app via a new user sign-up.

[LOOM DEMO](https://www.loom.com/share/747b7a8750d34bf594af7603b6e4b0b0)

### Steps To Reproduce
1. Use a public app [URL](https://app.appsmith.com/applications/62384cb64d9aea1b062ae5ee/pages/62384cb64d9aea1b062ae5f1) and open it up on incognito to be able to repro this scenarrio
2. Click on fork app and create a new user to access this app and observe the 400 error displayed on successfully creating the user (seen in the video)

Public app to fork: https://app.appsmith.com/applications/62384cb64d9aea1b062ae5ee/pages/62384cb64d9aea1b062ae5f1

=================================================================================
**Original issue description**

When a public app is forked using a new user, there's no edit button available and the only option available to the user is to navigate to the home page and the forked app does not exist in the organization

[LOOM DEMO](https://www.loom.com/share/63f89d54c2924cb496e6fcc77bcc1a30)

### Steps To Reproduce

1. Use templates and create an app on release URL. Copy from `application/...` and paste it after `https://appsmith-git-feat-betterurls-get-appsmith.vercel.app/`
4. Allow forking this app and make this app public
5. Open up the URL using a new user and observe there's no other option apart from navigating to the home page and once on the home page there is no card for the app that was just forked.

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by akash-codemonk**
## Description

When the fork button was clicked on by a non signed in user, after redirecting to the same page after sign up the fork modal wasn't being displayed. This was because the component did not exist in the header as in there is no route defined for the fork app url. The fix is to just add the route for the same. 

Fixes #11745 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Cypress test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/fork-button 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 56.35 **(0.01)** | 37.81 **(0.01)** | 35.98 **(0.02)** | 56.58 **(0)**
 :green_circle: | app/client/src/actions/applicationActions.ts | 74.55 **(1.82)** | 100 **(0)** | 22.22 **(5.55)** | 67.65 **(2.94)**
 :green_circle: | app/client/src/constants/routes.ts | 96.1 **(0.1)** | 100 **(0)** | 75 **(8.33)** | 95.89 **(0.06)**
 :green_circle: | app/client/src/pages/Applications/ForkApplicationModal.tsx | 84.78 **(1.45)** | 33.33 **(1.51)** | 44.44 **(6.94)** | 84.09 **(1.59)**</details>

### CODE REVIEW

**Review by arunvjn — DISMISSED** [review]

**Comment by arunvjn** [comment]
forkNonSignedInUser.json is over 6000 lines. 
Can we use a smaller import file here ?

**Comment by akash-codemonk** [comment]
> forkNonSignedInUser.json is over 6000 lines. Can we use a smaller import file here ?

Exported a forkable app and used it here. Will check if I can just update a flag in the json of a simple app to make it forkable.

**Review by arunvjn — DISMISSED** [review]

**Review by arunvjn — APPROVED** [review]

File: app/client/src/pages/Applications/ForkApplicationModal.tsx [hunk 1]
```diff
@@ -44,6 +45,12 @@ function ForkApplicationModal(props: ForkApplicationModalProps) {
     (state: AppState) => state.ui.applications.forkingApplication,
   );
 
+  useEffect(() => {
+    if (!userOrgs.length) {
+      dispatch(getAllApplications());
```
Comment (arunvjn) [inline app/client/src/pages/Applications/ForkApplicationModal.tsx:50]: Why do we need this API call ?
"/new" API should get us all the info we need right ?

Comment (akash-codemonk) [reply app/client/src/pages/Applications/ForkApplicationModal.tsx:50]: Yeah `getAllApplications` calls the /new api. This api doesn't get called in view mode. This is required in the fork modal to get the organization list to fork to.

**Pull request merged**